File: CS/MultiGroupSummary/MyGridControl/MyGridFooterMenu.cs
using DevExpress.XtraGrid.Menu;
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid;

namespace Q354185.MyXtraGrid {
    public class MyGridFooterMenu : GridViewFooterMenu {
        public MyGridFooterMenu(GridView view)
            : base(view) {
        }
        public override void Init(object info) {
            var myInfo = info as GridHitInfo;
            base.Init(info);
            var rowSummaryItem = View.GetRowSummaryItem(myInfo.RowHandle, myInfo.Column);
            SetSummaryItem((GridGroupSummaryItem)rowSummaryItem.Key);
            if (myInfo.HitTest != GridHitTest.RowFooter) {
                return;
            }
            var item = myInfo.FooterCell.ColumnInfo.Tag as GridGroupSummaryItem;
            SetSummaryItem(item);
        }
        private void SetSummaryItem(GridGroupSummaryItem item) {
            if (item == null) {
                return;
            }
            SummaryItem = item;
            CreateItems();
        }
    }
}

File: VB/MultiGroupSummary/MyGridControl/MyGridFooterMenu.vb
Imports DevExpress.XtraGrid.Menu
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid

Namespace Q354185.MyXtraGrid
    Public Class MyGridFooterMenu
        Inherits GridViewFooterMenu

        Public Sub New(ByVal view As GridView)
            MyBase.New(view)
        End Sub
        Public Overrides Sub Init(ByVal info As Object)
            Dim myInfo = TryCast(info, GridHitInfo)
            MyBase.Init(info)
            Dim rowSummaryItem = View.GetRowSummaryItem(myInfo.RowHandle, myInfo.Column)
            SetSummaryItem(CType(rowSummaryItem.Key, GridGroupSummaryItem))
            If myInfo.HitTest <> GridHitTest.RowFooter Then
                Return
            End If
            Dim item = TryCast(myInfo.FooterCell.ColumnInfo.Tag, GridGroupSummaryItem)
            SetSummaryItem(item)
        End Sub
        Private Sub SetSummaryItem(ByVal item As GridGroupSummaryItem)
            If item Is Nothing Then
                Return
            End If
            SummaryItem = item
            CreateItems()
        End Sub
    End Class
End Namespace
